TLM/TLM/Manager/Impl/LaneConnectionManagerData/ConnectionData.cs

```diff
@@ -0,0 +1,159 @@
+namespace TrafficManager.Manager.Impl.LaneConnectionManagerData {
+    using CSUtil.Commons;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using TrafficManager.Util;
+    using TrafficManager.Util.Extensions;
+    using static TrafficManager.Util.Shortcuts;
+    internal struct LaneConnectionData {
+        public uint LaneId;
+        // TODO [issue 354]: add track/car connection type.
+
+        public LaneConnectionData(uint laneId) {
+            LaneId = laneId;
+        }
+
+        public override string ToString() => $"TargetConnectionData({LaneId})";
+    }
+
+    /// <summary>
+    /// All calls to this class assumes lanes are valid
+    /// </summary>
+    internal class ConnectionData {
+        private SourceLaneConnectionData[] connections_;
+
+        public ConnectionData() {
+            connections_ = new SourceLaneConnectionData[NetManager.instance.m_lanes.m_size];
```

Comment:
so this is essentially half connection (collection of starting points), right?

Reply:
half? connections are uni-directional so I don't know why do you use the word half.
it is an where index is starting point and element value contains the target point.

Reply:
by pointer you mean array?
what do you mean by unsafe? I am not using unsafe block. why is this unsafe? I don't see any problem.